libkbfs/bserver_memory.go

```diff
@@ -175,6 +175,52 @@ func (b *BlockServerMemory) Put(ctx context.Context, tlfID tlf.ID,
 	return refs.put(context, liveBlockRef, "")
 }
 
+// PutAgain implements the BlockServer interface for BlockServerMemory.
+func (b *BlockServerMemory) PutAgain(ctx context.Context, tlfID tlf.ID,
+	id kbfsblock.ID, context kbfsblock.Context, buf []byte,
+	serverHalf kbfscrypto.BlockCryptKeyServerHalf) (err error) {
+
+	if err := checkContext(ctx); err != nil {
+		return err
+	}
+
+	defer func() {
+		err = translateToBlockServerError(err)
+	}()
+	b.log.CDebugf(ctx, "BlockServerMemory.PutAgain id=%s tlfID=%s context=%s "+
+		"size=%d", id, tlfID, context, len(buf))
+	b.lock.Lock()
```

Comment:
Everything from this point on looks copied from `BlockServerMemory.Put()` (except this is missing the key half check).  Can we refactor it into a shared helper function?